apps/frontend/core-dapp/src/components/SettingsMenu/RpcSettings.tsx

```diff
@@ -0,0 +1,88 @@
+import { useState } from 'react'
+import { observer } from 'mobx-react-lite'
+import { Button, Flex, Icon, Input, spacingIncrement } from 'prepo-ui'
+import styled from 'styled-components'
+import SettingsMenuItem from './SettingsMenuItem'
+import { useRootStore } from '../../context/RootStoreProvider'
+import { isProduction } from '../../utils/isProduction'
+
+const FormWrapper = styled.div<{ show: boolean }>`
+  background-color: ${({ theme }): string => theme.color.neutral10};
+  height: 100%;
+  left: ${({ show }): string => (show ? '0' : '100%')};
+  padding: ${spacingIncrement(16)};
+  position: absolute;
+  top: 0;
+  transition: 0.2s;
+  width: 100%;
+`
+
+const CloseIconWrapper = styled.div`
+  color: ${({ theme }): string => theme.color.neutral2};
+  cursor: pointer;
+  :hover {
+    opacity: 0.7;
+  }
+`
+
+const Description = styled.p`
+  color: ${({ theme }): string => theme.color.neutral3};
+`
+
+const Title = styled.p`
+  color: ${({ theme }): string => theme.color.secondary};
+  font-size: ${({ theme }): string => theme.fontSize.base};
+  font-weight: ${({ theme }): number => theme.fontWeight.semiBold};
+`
+
+const RpcSettings: React.FC<{ onShowForm: () => void }> = () => {
+  const { web3Store } = useRootStore()
+  const { customRpc, network } = web3Store
+
+  const [showForm, setShowForm] = useState(false)
+  const [rpcUrl, setRpcUrl] = useState(customRpc[network.name] ?? '')
+  const disabled = customRpc[network.name] === rpcUrl || rpcUrl === ''
+
+  if (isProduction) return null
+
+  const handleSave = (): void => {
+    web3Store.setRpc(network.name, rpcUrl)
+  }
+
+  return (
+    <>
+      <SettingsMenuItem
+        iconName="chevron-right"
+        onClick={() => {
+          setShowForm(true)
+        }}
+      >
+        RPC Settings
+      </SettingsMenuItem>
+      <FormWrapper show={showForm}>
+        <Flex alignItems="center" justifyContent="space-between" mb={12}>
+          <Title>RPC Settings</Title>
+          <CloseIconWrapper onClick={() => setShowForm(false)}>
+            <Icon name="cross" height="16" width="16" />
+          </CloseIconWrapper>
+        </Flex>
+        <Flex gap={12} flexDirection="column">
+          <Description>Entry custom RPC for {network.displayName ?? network.chainName}</Description>
+          <Input
+            size="small"
+            placeholder="e.g. https://rpc.custom.com"
+            value={rpcUrl}
+            onChange={(e) => {
+              setRpcUrl(e.target.value)
+            }}
+          />
```

Comment:
Some suggestions:

* Shorten the copy:

    ```suggestion
              <Description>Custom RPC for {network.displayName ?? network.chainName}</Description>
              <Input
                size="small"
                placeholder="e.g. https://rpc.custom.com"
                value={rpcUrl}
                onChange={(e) => {
                  setRpcUrl(e.target.value)
                }}
              />
    ```

* Make `Description` a label and attach it to the input for better accessibility.

Reply:
The label in Input is too big and I don't want to go through the hassle of changing the Input component since this UI may eventually be completely different anyway, shortened the label